src/network/networkd-dhcp4.c

```diff
@@ -273,7 +273,8 @@ static int dhcp4_address_handler(sd_netlink *rtnl, sd_netlink_message *m,
         } else if (r >= 0)
                 manager_rtnl_process_address(rtnl, m, link->manager);
 
-        link_set_dhcp_routes(link);
+        if (link->network->dhcp_use_routes)
+                link_set_dhcp_routes(link);
```

Comment:
I think it would be nicer to move the if check into link_set_dhcp_routes(). i.e. check the bool there, and exit early if it isn't set.